crates/fuel-core/src/graphql_api/logging.rs

```diff
@@ -0,0 +1,74 @@
+use async_graphql::{
+    extensions::{
+        Extension,
+        ExtensionContext,
+        ExtensionFactory,
+        NextRequest,
+        NextResolve,
+        ResolveInfo,
+    },
+    Response,
+    ServerResult,
+    Value,
+};
+use std::{
+    sync::Arc,
+    time::Duration,
+};
+use tokio::time::Instant;
+
+pub(crate) struct LoggingExtension {
```

Comment:
I like the idea of combining `LoggingExtension` and `PrometheusExtension` into `LogsAndMetricsExtension` or `MetricsExtension`. In this case, you will spawn only one `Instance`=)

Reply:
I went with separate because combining them feels like a violation of the [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single-responsibility_principle).

If performance with `Instance` is a big concern, I'm happy to reconsider. But I just default to architecturally cleanliness over performance.

Reply:
The logging of long queries is more about metrics than logs. Plus, we have a separate `Tracing` extension for logs=)

Yeah, it is not integrated with Prometheus, but if we rename the extension into `MetricsExtension`, then it would be ergonomic=)

<img width="807" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/53d0de32-15f7-4b06-862a-a0f4046855f6">